evcache-core/src/main/java/com/netflix/evcache/pool/EVCacheClientUtil.java

```diff
@@ -25,27 +25,26 @@ public EVCacheClientUtil(EVCacheClientPool pool) {
         this._appName = pool.getAppName();
     }
 
+    //TODO: Remove this todo. This method has been made hashing agnostic.
     /**
      * TODO : once metaget is available we need to get the remaining ttl from an existing entry and use it 
      */
-    public EVCacheLatch add(EVCacheKey evcKey, final CachedData cd, boolean shouldHashKey, Transcoder evcacheValueTranscoder, int timeToLive, Policy policy) throws Exception {
+    public EVCacheLatch add(EVCacheKey evcKey, final CachedData cd, Transcoder evcacheValueTranscoder, int timeToLive, Policy policy) throws Exception {
         if (cd == null) return null; 
         
         final EVCacheClient[] clients = _pool.getEVCacheClientForWrite();
         final EVCacheLatchImpl latch = new EVCacheLatchImpl(policy, clients.length - _pool.getWriteOnlyEVCacheClients().length, _appName);
 
-        CachedData cd1 = null;
```

Comment:
@smadappa I had to pull this variable down, to the level of each client, since otherwise, this can have incorrect influence on other clients.
For instance, if the first client in the for loop has hashing enabled, cd1 contains its hashedkey. Now, for the subsequent client, if hashing is not enabled, we will still end up using cd1 of the previous client.

Reply:
Looks good. The only thing is we might encode twice in some situations for add esp if we have mixed algo or some copies hashed and some not hashed.

Reply:
This might be ok as I don't expect this to be a common use case. This might be pretty rare if it were to even occur. We can address this if it happens.

Reply:
@smadappa I think I see what you mean. We will not be double encoding or double hashing, but we may be repeating the work we already did of calculating the encoded value, for each of the clients. Yeah, let us revisit this if needed, and as a potential improvement, we can explore storing each encoded value (each for non-duet and duet) and reuse if needed.